app/[locale]/projects/page.tsx

```diff
@@ -42,12 +42,23 @@ const ProjectsPage = () => {
     setLoading(true);
     const existingTags = structuredClone(tags);
 
-    // we set tag active to true
-    for (const exisitingTag of existingTags) {
+    // we set tags chosen to active and all the others to not
+    if (existingTags.every(value => value.isActive)) {
+      existingTags.forEach(existingTag => {
+        existingTag.isActive = false;
+      });
+    }
+
+    existingTags.forEach(exisitingTag => {
       if (exisitingTag.name === tag.name) {
         exisitingTag.isActive = !exisitingTag.isActive;
-        break;
       }
+    });
+
+    if (existingTags.every(value => !value.isActive)) {
+      existingTags.forEach(existingTag => {
```

Comment:
You can save the `existingTags.every(value => !value.isActive` from before into a boolean and check it in here (you would probably need to check for !yourBooleanValue)

And you will save one array function